File: Temperature/Github-Copilot/Python/Temperature1.py
#Temperature measurements are taken in one and the same place over a number of weeks. The measurements are taken a fixed number of times, the same number of measurements in each week.  At the end of the measurement period, the collected data must be processed: for each week, the smallest, the largest and the average temperature is determined. The minimum, maximum and average temperature must also be determined for the entire measurement period. All meassurements are decimal numbers with 1 point precision.
#The input data is given through standard input on the following format: The first row contains the number of weeks and the number of measurements per week separated by spaces. The following rows contain the measurements for each week, one row per week, meassurements separated by spaces.
#The output data should be to standard output given in the following format: One row for each week containing the smallest, largest and average to 1 point precision temperature separated by spaces. One row containing the minimum, maximum and averageto 1 point precision temperature for the entire measurement period separated by spaces.

def main():
    # Read the input data
    num_weeks, num_measurements = map(int, input().split())
    measurements = []
    for _ in range(num_weeks):
        measurements.append(list(map(float, input().split())))

    # Process the data
    min_temp = max_temp = sum_temp = measurements[0][0]
    min_temps = []
    max_temps = []
    sum_temps = []
    for week in measurements:
        min_temp = min(week)
        max_temp = max(week)
        sum_temp = sum(week)
        min_temps.append(min_temp)
        max_temps.append(max_temp)
        sum_temps.append(sum_temp)
        print("{:.1f} {:.1f} {:.1f}".format(min_temp, max_temp, sum_temp / num_measurements))

    print("{:.1f} {:.1f} {:.1f}".format(min(min_temps), max(max_temps), sum(sum_temps) / (num_weeks * num_measurements)))
    

if __name__ == "__main__":
    main()

File: Temperature/CodePal/Java/Temperature1.java
import java.util.Scanner;

public class Temperature1 {
    /**
     * This function takes temperature measurements in one place over a number of
     * weeks and processes the data to determine
     * the smallest, largest, and average temperature for each week and for the
     * entire measurement period.
     * 
     * @param numWeeks               The number of weeks of temperature measurements
     * @param numMeasurementsPerWeek The number of measurements taken each week
     * @param temperatureData        A 2D array containing the temperature
     *                               measurements for each week
     * @return A string containing the smallest, largest, and average temperature
     *         for each week and for the entire measurement period
     */
    public static String processTemperatureData(int numWeeks, int numMeasurementsPerWeek, double[][] temperatureData) {
        // Initialize variables to hold the minimum, maximum, and average temperature
        // for each week and for the entire measurement period
        double[] minTemps = new double[numWeeks];
        double[] maxTemps = new double[numWeeks];
        double[] avgTemps = new double[numWeeks];
        double overallMin = Double.MAX_VALUE;
        double overallMax = Double.MIN_VALUE;
        double overallSum = 0.0;

        // Loop through each week of temperature measurements
        for (int i = 0; i < numWeeks; i++) {
            double minTemp = Double.MAX_VALUE;
            double maxTemp = Double.MIN_VALUE;
            double sum = 0.0;

            // Loop through each measurement in the week and determine the minimum, maximum,
            // and sum of temperatures
            for (int j = 0; j < numMeasurementsPerWeek; j++) {
                double temp = temperatureData[i][j];
                if (temp < minTemp) {
                    minTemp = temp;
                }
                if (temp > maxTemp) {
                    maxTemp = temp;
                }
                sum += temp;
            }

            // Calculate the average temperature for the week
            double avgTemp = sum / numMeasurementsPerWeek;

            // Store the minimum, maximum, and average temperature for the week
            minTemps[i] = minTemp;
            maxTemps[i] = maxTemp;
            avgTemps[i] = avgTemp;

            // Update the overall minimum, maximum, and sum of temperatures
            if (minTemp < overallMin) {
                overallMin = minTemp;
            }
            if (maxTemp > overallMax) {
                overallMax = maxTemp;
            }
            overallSum += sum;
        }

        // Calculate the overall average temperature
        double overallAvg = overallSum / (numWeeks * numMeasurementsPerWeek);

        // Construct the output string
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < numWeeks; i++) {
            sb.append(String.format("%.1f %.1f %.1f\n", minTemps[i], maxTemps[i], avgTemps[i]));
        }
        sb.append(String.format("%.1f %.1f %.1f", overallMin, overallMax, overallAvg));

        return sb.toString();
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int numWeeks = scanner.nextInt();
        int numMeasurements = scanner.nextInt();
        double[][] measurements = new double[numWeeks][numMeasurements];
        for (int i = 0; i < numWeeks; i++) {
            for (int j = 0; j < numMeasurements; j++) {
                measurements[i][j] = scanner.nextDouble();
            }
        }
        scanner.close();

        System.out.println(processTemperatureData(numWeeks, numMeasurements, measurements));
    }
}
